lib/common/src/RemoteRequest/CurlRemoteRequest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Amp\RemoteRequest;
+
+use Amp\Exception\FailedToFetchFromRemoteUrl;
+use Amp\RemoteRequest;
+use Exception;
+
+/**
+ * Remote request transport using cURL.
+ *
+ * @package amp/common
+ */
+final class CurlRemoteRequest implements RemoteRequest
```

Comment:
Maybe we can use `get()` instead of `fetch()`, and can always add `post()` or other methods if needed.

Reply:
I thought through this, and we should have different interfaces for different method. Otherwise, we'd need to add all possible methods to the one interface right from the start, to avoid changing the interface later on and breaking existing implementation of the interface.

Therefore, I'll rename `RemoteRequest` into `RemoteGetRequest`. We can still have a single implementation serve all interfaces though, it's just that we have "segregated interfaces", i.e.: `final class CurlRemoteRequest implements RemoteGetRequest, RemotePostRequest`.